### PR TITLE
Feature 5451 - Ban adherent feature

### PR DESCRIPTION
**Pull request by tony-tran**

### CODE REVIEW

**Comment by thomasbisignani** [comment]
You should add also the new role `ROLE_ADMIN_BAN` in the AdministratorAdmin et the Admin fixtures : 

https://github.com/EnMarche/en-marche.fr/blob/master/src/Admin/AdministratorAdmin.php#L106
https://github.com/EnMarche/en-marche.fr/blob/master/src/DataFixtures/ORM/LoadAdminData.php#L58

File: src/Validator/AdherentBannedValidator.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+    private $objectManager;
+
+    public function __construct(ObjectManager $objectManager)
+    {
+        $this->objectManager = $objectManager;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
```
Comment (thomasbisignani) [inline src/Validator/AdherentBannedValidator.php]: You should also check the constraint `AdherentBanned` instance and throw a `UnexpectedTypeException`
Example :
```php
        if (!$constraint instanceof AdherentBanned) {
            throw new UnexpectedTypeException($constraint, AdherentBanned::class);
        }
```

File: src/Controller/Admin/AdminAdherentController.php [hunk 2]
```diff
@@ -28,4 +33,30 @@ public function exitImpersonationAction(): Response
 
         return $this->redirectToRoute('admin_app_adherent_list');
     }
+
+    /**
+     * @Route("adherent/{id}/ban", name="app_admin_ban_adherent")
```
Comment (ottaviano) [inline src/Controller/Admin/AdminAdherentController.php]: can prefix by `/` the URI?

File: templates/admin/adherent/edit.html.twig [hunk 3]
```diff
@@ -1,6 +1,13 @@
 {% extends 'SonataAdminBundle:CRUD:base_edit.html.twig' %}
 
 {% block actions %}
+    <script>
+        $(function(){
+            $('.ban-link').click(function(){
+                $('#ban').submit();
```
Comment (ottaviano) [inline templates/admin/adherent/edit.html.twig]: why do you not use `<input type=submit>` button?

Comment (tony-tran) [reply templates/admin/adherent/edit.html.twig]: I prefer to keep the `<a href>` to have the same design of links.
If I add the `input` I need to add the click event on this input and IMHO it's no necessary. I prefer to submit directly the form. WDYT ?

**Pull request merged**